packages/authentication/test/unit/metadata-provider.ts

```diff
@@ -0,0 +1,74 @@
+// Copyright IBM Corp. 2017. All Rights Reserved.
+// Node module: loopback
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {expect} from '@loopback/testlab';
+import {Context, Provider} from '@loopback/context';
+import {ParsedRequest} from '@loopback/core';
+import {AuthMetadataProvider, IMetadata, authenticate} from '../..';
+
+describe('AuthMetadataProvider', () => {
+  let provider: Provider<IMetadata>;
+
+  class TestController {
+    @authenticate('my-strategy', {option1: 'value1', option2: 'value2'})
+    async whoAmI() {}
+  }
+
+  class ControllerWithNoMetadata {
+    async whoAmI() {}
+  }
+
+  const testController: TestController = new TestController();
+  const controllerNoMetadata: ControllerWithNoMetadata = new ControllerWithNoMetadata();
+
+  beforeEach(givenAuthMetadataProvider);
+
+  describe('value()', () => {
+    it('returns the authentication metadata of a controller method', async () => {
+       const authMetadata: IMetadata = await Promise.resolve(provider.value());
```

Comment:
the generic Provider.value() interface method in the core package has the return value as ValueOrPromise, so though we know that the implemented class in authentication package returns a Value, the Provider.value() return value has to be resolved to a value.